testcloud/instance.py

```diff
@@ -278,6 +278,34 @@ def prepare(self, data_tpl=None):
         # create the dirs needed for this instance
         try:
             self._create_dirs()
+            if self.domain_configuration.console_log_file:
+                # SELinux is quite picky where it permits virtlogd to write logs,
+                # ConfigData.CONSOLE_LOG_DIR must point to location where virtlogd
+                # is permitted by SELinux to search, create the actual log file
+                # in the CONSOLE_LOG_DIR and point symlink to it
+                console_log_real_path = get_console_log_real_path(str(self.domain_configuration.uuid))
+                log.debug("Creating console log symlink {} -> {}".format(self.domain_configuration.console_log_file, console_log_real_path))
+                os.symlink(console_log_real_path, self.domain_configuration.console_log_file)
```

Comment:
An alternative approach came to my mind: Caller would just state that they are interested in keeping the console log:

```python
domain_configuration.keep_console_log = True
```

And then do whatever they need with the file path provided in `domain_configuration.console_log_file`. However, the symlink approach is completely fine from my point of view as well.